### PR TITLE
useMouse

### LINKED ISSUE #47: useMouse
Port mouse sensor: https://github.com/streamich/libreact/blob/master/docs/en/MouseSensor.md

### PR DESCRIPTION
**Pull request by wardoost**
Hook that tracks mouse position, ported from the libreact's [`MouseSensor`](https://github.com/streamich/libreact/blob/master/docs/en/MouseSensor.md). Still have to add the `whenHovered` option though. I'm also not a 100% sold on the naming of the returned state variables. I think `posX` and `posY` could just be `x` and `y`?

Closes #47

### CODE REVIEW

**Comment by wardoost** [comment]
What about a `useWindowMouse` hook that only returns `x` and `y` coords relative to the window? Could also be done when not passing in a ref in this hook

File: src/useMouse.ts [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+    }
+
+    document.addEventListener('mousemove', handler);
+
+    return () => {
+      if (frame.current) {
+        cancelAnimationFrame(frame.current);
+      }
+
+      document.addEventListener('mousemove', handler);
```
Comment (streamich) [inline src/useMouse.ts]: `removeEventListener`

File: src/useMouse.ts [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+    posY: 0,
+    elX: 0,
+    elY: 0,
+    elH: 0,
+    elW: 0,
+  });
+
+  useEffect(() => {
+    const handler = (event: MouseEvent) => {
+      frame.current = requestAnimationFrame(() => {
```
Comment (streamich) [inline src/useMouse.ts]: Should there be 

```js
cancelAnimationFrame(frame.current)
```

before this line?

**Comment by streamich** [comment]
`useWindowMouse` would be useful, too. I don't know if it is better to put it inside this hook, or make a separate one.

File: src/useMouse.ts (hunk 1 above)
Comment (wardoost) [reply src/useMouse.ts]: Fixed

File: src/useMouse.ts (hunk 2 above)
Comment (wardoost) [reply src/useMouse.ts]: Yeah, I think so. I noticed I didn't put in above [this line](https://github.com/streamich/react-use/blob/master/src/useScroll.ts#L19) in `useScroll` either 😱

**Pull request merged**